packages/sdk/src/utils/index.ts

```diff
@@ -0,0 +1,320 @@
+  return parseInt((btc * 10 ** 8).toString()); // remove floating point overflow by parseInt
+}
+
+export function generateTxUniqueIdentifier(txId: string, index: number) {
+  return `${txId}:${index}`;
+}
+
+export function decodePSBT({ hex, base64, buffer }: OneOfAllDataFormats): Psbt {
+  if (hex) return Psbt.fromHex(hex);
+  if (base64) return Psbt.fromBase64(base64);
+  if (buffer) return Psbt.fromBuffer(buffer);
+
+  throw new OrditSDKError("Invalid options");
+}
+
+export function decodeTx({ hex, buffer }: BufferOrHex): Transaction {
+  if (hex) return Transaction.fromHex(hex);
+  if (buffer) return Transaction.fromBuffer(buffer);
+
+  throw new OrditSDKError("Invalid options");
+}
+
+function isPaymentFactory(payment: PaymentCreator, network: Network) {
+  return (script: Buffer) => {
+    try {
+      return payment({ output: script, network: getNetwork(network) });
+    } catch (error) {
+      return false;
+    }
+  };
+}
+
+export const isP2PKH = (
+  script: Buffer,
+  network: Network,
+): IsBitcoinPaymentResponse => {
+  const p2pkh = isPaymentFactory(payments.p2pkh, network)(script);
+  return {
+    type: "p2pkh",
+    payload: p2pkh,
+  };
+};
+export const isP2WPKH = (
+  script: Buffer,
+  network: Network,
+): IsBitcoinPaymentResponse => {
+  const p2wpkh = isPaymentFactory(payments.p2wpkh, network)(script);
+  return {
+    type: "p2wpkh",
+    payload: p2wpkh,
+  };
+};
+export const isP2WSHScript = (
+  script: Buffer,
+  network: Network,
+): IsBitcoinPaymentResponse => {
+  const p2wsh = isPaymentFactory(payments.p2wsh, network)(script);
+  return {
+    type: "p2sh",
+    payload: p2wsh,
+  };
+};
+export const isP2SHScript = (
+  script: Buffer,
+  network: Network,
+): IsBitcoinPaymentResponse => {
+  const p2sh = isPaymentFactory(payments.p2sh, network)(script);
+  return {
+    type: "p2sh",
+    payload: p2sh,
+  };
+};
+export const isP2TR = (
+  script: Buffer,
+  network: Network,
+): IsBitcoinPaymentResponse => {
+  const p2tr = isPaymentFactory(payments.p2tr, network)(script);
+  return {
+    type: "p2tr",
+    payload: p2tr,
+  };
+};
+export function getScriptType(
+  script: Buffer,
+  network: Network,
+): GetScriptTypeResponse {
+  const p2pkh = isP2PKH(script, network);
+  if (p2pkh.payload) {
+    return {
+      format: ADDRESS_TYPE_TO_FORMAT["p2pkh"],
+      ...p2pkh,
+    };
+  }
+
+  const p2wpkh = isP2WPKH(script, network);
+  if (p2wpkh.payload) {
+    return {
+      format: ADDRESS_TYPE_TO_FORMAT["p2wpkh"],
+      ...p2wpkh,
+    };
+  }
+
+  const p2sh = isP2SHScript(script, network);
+  if (p2sh.payload) {
+    return {
+      format: ADDRESS_TYPE_TO_FORMAT["p2sh"],
+      ...p2sh,
+    };
+  }
+
+  const p2tr = isP2TR(script, network);
+  if (p2tr.payload) {
+    return {
+      format: ADDRESS_TYPE_TO_FORMAT["p2tr"],
+      ...p2tr,
+    };
+  }
+
+  throw new OrditSDKError("Unsupported input");
+}
+
+export function getDummyP2TRInput(): UTXO {
```

Comment:
@iamcrazycoder what is this used for?

Reply:
Its used to calculate network fee in preview mode w\ needing a confirmed utxo. Only used in the inscription flow where we need to know the reveal deposit + network for the user to make an accurate deposit.

Check `Inscriber` class to see how its used.